src/base/dof_map.C

```diff
@@ -254,17 +254,23 @@ void DofMap::set_error_on_constraint_loop(bool error_on_constraint_loop)
 
 void DofMap::add_variable_group (const VariableGroup & var_group)
 {
-  const unsigned int vg = cast_int<unsigned int>(_variable_groups.size());
+  // Ensure that we are not duplicating an existing entry in _variable_groups
+  if (std::find(_variable_groups.begin(), _variable_groups.end(), var_group) == _variable_groups.end())
```

Comment:
I was going to suggest that 
```
if (!std::any_of(_variable_groups.begin(), _variable_groups.end(), std::equal_to(var_group)))
```
was terser while doing the same amount of comparisons, but after typing it all out I'm not sure. I do think it reads nicer than "find == end" in general though.

Reply:
I am definitely not up to speed with the latest C++ conventions. I would have thought that the find statement is more readable. I feel the same way about make_range actually. However, if none_of is more efficient/in tune with standards, I can change this.

Reply:
>  latest C++ conventions

I'd make fun of you because `std::none_of` is C++11, but I'd forgot it existed until John pointed it out...

I prefer find() but I don't really like any of the options C++ standards give us here.  I want to just call a vector::contains(), but we don't even get set::contains() until C++20, nor string::contains() until C++23, and still no std::contains() in sight?

Reply:
Yeah, the STL always uses ranges because that's the most general thing to do... but it ain't that pretty sometimes.